### PR TITLE
Additional Core Performance Improvements

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

- Adds Iterator implementation for iteration of Netty HttpHeaders.
- Defers creation of StringBuilder until it is needed.
- Specializes `UrlTokenizer`'s reading until a given character.
- Adds JFR configuration file

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-http-okhttp/src/main/java/module-info.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
     requires okhttp3;
     requires okio;
+    requires kotlin.stdlib;
```
Comment (JonathanGiles) [inline sdk/core/azure-core-http-okhttp/src/main/java/module-info.java:9]: We should consider having an okhttp impl that doesn't bring in kotlin

**Review by JonathanGiles — APPROVED** [review]

File: sdk/core/azure-core-http-okhttp/src/main/java/module-info.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-http-okhttp/src/main/java/module-info.java:9]: Yeah, I was surprised this was needed when OkHttp's `Headers.forEach` was used

**Comment by alzimmermsft** [comment]
Transient test failure unrelated to this PR

**Pull request merged**